aws/resource_aws_emr_cluster.go

```diff
@@ -796,25 +1000,42 @@ func resourceAwsEMRClusterRead(d *schema.ResourceData, meta interface{}) error {
 
 	instanceGroups, err := fetchAllEMRInstanceGroups(emrconn, d.Id())
 
-	if err != nil {
-		return err
-	}
+	if err == nil { // find instance group
 
-	coreGroup := emrCoreInstanceGroup(instanceGroups)
-	masterGroup := findMasterGroup(instanceGroups)
+		coreGroup := emrCoreInstanceGroup(instanceGroups)
+		masterGroup := findMasterGroup(instanceGroups)
 
-	flattenedCoreInstanceGroup, err := flattenEmrCoreInstanceGroup(coreGroup)
+		flattenedCoreInstanceGroup, err := flattenEmrCoreInstanceGroup(coreGroup)
 
-	if err != nil {
-		return fmt.Errorf("error flattening core_instance_group: %s", err)
-	}
+		if err != nil {
+			return fmt.Errorf("error flattening core_instance_group: %s", err)
+		}
 
-	if err := d.Set("core_instance_group", flattenedCoreInstanceGroup); err != nil {
-		return fmt.Errorf("error setting core_instance_group: %s", err)
+		if err := d.Set("core_instance_group", flattenedCoreInstanceGroup); err != nil {
+			return fmt.Errorf("error setting core_instance_group: %s", err)
+		}
+
+		if err := d.Set("master_instance_group", flattenEmrMasterInstanceGroup(masterGroup)); err != nil {
+			return fmt.Errorf("error setting master_instance_group: %s", err)
+		}
 	}
 
-	if err := d.Set("master_instance_group", flattenEmrMasterInstanceGroup(masterGroup)); err != nil {
-		return fmt.Errorf("error setting master_instance_group: %s", err)
+	instanceFleets, err := fetchAllEMRInstanceFleets(emrconn, d.Id())
+
+	if err == nil { // find instance fleets
+
+		coreFleet := findInstanceFleet(instanceFleets, emr.InstanceRoleTypeCore)
+		masterFleet := findInstanceFleet(instanceFleets, emr.InstanceRoleTypeMaster)
```

Comment:
Nit: Since the API includes an enumeration type specifically for instance fleets, we should use those values just to reduce any potential confusion. 👍 

```suggestion
		coreFleet := findInstanceFleet(instanceFleets, emr.InstanceFleetTypeCore)
		masterFleet := findInstanceFleet(instanceFleets, emr.InstanceFleetTypeMaster)
```